### PR TITLE
auth: add quiet modifier to pdnsutil rectify-all-zones command

### LINKED ISSUE #6245: Please make rectify-all-zones cronjob friendly
For distribution packages it'd be nice if we could ship a daily cronjob that runs "pdnsutil rectify-all-zones". This should however not print any output or exit non-zero when there are no actionable items.

### PR DESCRIPTION
**Pull request by chbruyand**
### Short description
Add quiet modifier to pdnsutil rectify-all-zones command to only output on error.
Also properly exit with a non-zero code when an error is encountered.

Fix #6245 

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/pdnsutil.cc [hunk 1]
```diff
@@ -1905,7 +1916,7 @@ try
     cout<<"                                   List all zone names"<<endl;;
     cout<<"list-tsig-keys                     List all TSIG keys"<<endl;
     cout<<"rectify-zone ZONE [ZONE ..]        Fix up DNSSEC fields (order, auth)"<<endl;
-    cout<<"rectify-all-zones                  Rectify all zones."<<endl;
+    cout<<"rectify-all-zones [quiet]          Rectify all zones. Optionnaly quiet output with errors only"<<endl;
```
Comment (zeha) [inline pdns/pdnsutil.cc]: typo in optionally

File: pdns/pdnsutil.cc [hunk 2]
```diff
@@ -202,17 +205,25 @@ void dbBench(const std::string& fname)
...
+    }
+    if (!rectifyZone(dk, di.zone, quiet)) {
+      result = false;
+    }
+  }
+  if (!quiet) {
+    cout<<"Rectified "<<domainInfo.size()<<" zones."<<endl;
   }
-  cout<<"Rectified "<<domainInfo.size()<<" zones."<<endl;
+  return (result);
```
Comment (rgacogne) [inline pdns/pdnsutil.cc]: I'm not a big fan of the parentheses here.

File: pdns/pdnsutil.cc (hunk 1 above)
Comment (rgacogne) [inline pdns/pdnsutil.cc]: s/Optionnaly/Optionally/

File: pdns/pdnsutil.cc [hunk 3]
```diff
@@ -1905,7 +1916,7 @@ try
     cout<<"                                   List all zone names"<<endl;;
     cout<<"list-tsig-keys                     List all TSIG keys"<<endl;
     cout<<"rectify-zone ZONE [ZONE ..]        Fix up DNSSEC fields (order, auth)"<<endl;
-    cout<<"rectify-all-zones                  Rectify all zones."<<endl;
+    cout<<"rectify-all-zones [quiet]          Rectify all zones. Optionnally quiet output with errors only"<<endl;
```
Comment (pieterlexis) [inline pdns/pdnsutil.cc]: if you could replace 'optionnally' with 'optionally' I'd be grateful :)

Comment (chbruyand) [reply pdns/pdnsutil.cc]: Sure... My english is so bad :/

Comment (rgacogne) [reply pdns/pdnsutil.cc]: I'm sure we could hook up aspell in a pre-commit hook or something ;)

Comment (pieterlexis) [reply pdns/pdnsutil.cc]: > I'm sure we could hook up aspell in a pre-commit hook or something ;)


or in travis :laughing:

Comment (krionbsd) [reply pdns/pdnsutil.cc]: or in vim like:
set spell spelllang=en,ru,de

and put dictionaries from http://ftp.vim.org/vim/runtime/spell/ into ~/.vim/spell :-)

Comment (chbruyand) [reply pdns/pdnsutil.cc]: But who still uses vim ?

I mean, it's 2018...

**Pull request merged**